File: community/src/main/java/com/petcom/community/entity/Breeder.java
package com.petcom.community.entity;

import com.fasterxml.jackson.annotation.JsonAutoDetect;

import java.util.Date;
@JsonAutoDetect(fieldVisibility=JsonAutoDetect.Visibility.ANY, getterVisibility= JsonAutoDetect.Visibility.NONE)
public class Breeder {
    private int id;
    private String title;
    private int score;
    private int type; // 0-Breeder, 1-Shelter
    private String description;
    private Date createTime;
    private String address;
    private String city;
    private String state;
    private String contact;
    private String website;

    public int getStatus() {
        return status;
    }

    @Override
    public String toString() {
        return "Breeder{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", score=" + score +
                ", type=" + type +
                ", description='" + description + '\'' +
                ", createTime=" + createTime +
                ", address='" + address + '\'' +
                ", city='" + city + '\'' +
                ", state='" + state + '\'' +
                ", contact='" + contact + '\'' +
                ", website='" + website + '\'' +
                ", status=" + status +
                '}';
    }

    public void setStatus(int status) {
        this.status = status;
    }

    private int status;

    public String getWebsite() {
        return website;
    }

    public void setWebsite(String website) {
        this.website = website;
    }

    public String getContact() {
        return contact;
    }

    public void setContact(String contact) {
        this.contact = contact;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getScore() {
        return score;
    }

    public void setScore(int score) {
        this.score = score;
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getCreateTime() {
        return createTime;
    }

    public void setCreateTime(Date createTime) {
        this.createTime = createTime;
    }

}

File: lib/model/Breeder.dart
class BreederResponse {
  String? msg;
  int? code;
  int? totalPage;
  int? currentPage;
  int? total;
  List<Breeder>? breeder;

  BreederResponse(
      {this.msg,
      this.code,
      this.totalPage,
      this.total,
      this.currentPage,
      this.breeder});

  BreederResponse.fromJson(Map<String, dynamic> json) {
    msg = json['msg'];
    code = json['code'];
    totalPage = json['total_page'];
    currentPage = json['current_page'];
    total = json['total'];
    if (json['breeder'] != null) {
      breeder = [];
      json['breeder'].forEach((v) {
        breeder!.add(new Breeder.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['msg'] = this.msg;
    data['code'] = this.code;
    data['total_page'] = this.totalPage;
    data['current_page'] = this.currentPage;
    data['total'] = this.total;
    if (this.breeder != null) {
      data['breeder'] = this.breeder!.map((v) => v.toJson()).toList();
    }
    return data;
  }
}

class Breeder {
  String? address;
  String? city;
  String? contact;
  int? createTime;
  String? description;
  int? id;
  double? score;
  String? state;
  String? title;
  int? type;
  String? website;
  int? status;

  Breeder({
    this.address,
    this.city,
    this.contact,
    this.createTime,
    this.description,
    this.id,
    this.score,
    this.state,
    this.title,
    this.type,
    this.website,
    this.status,
  });

  Breeder.fromJson(Map<String, dynamic> json) {
    address = json['address'];
    city = json['city'];
    contact = json['contact'];
    createTime = json['createTime'];
    description = json['description'];
    id = json['id'];
    score = json['score'];
    state = json['state'];
    title = json['title'];
    type = json['type'];
    website = json['website'];
    status = json['status'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['address'] = this.address;
    data['city'] = this.city;
    data['contact'] = this.contact;
    data['createTime'] = this.createTime;
    data['description'] = this.description;
    data['id'] = this.id;
    data['score'] = this.score;
    data['state'] = this.state;
    data['title'] = this.title;
    data['type'] = this.type;
    data['website'] = this.website;
    data['status'] = this.status;

    return data;
  }
}
